lib/broadway_dashboard/metrics.ex

```diff
@@ -85,31 +119,24 @@ defmodule BroadwayDashboard.Metrics do
   end
 
   @impl true
-  def handle_call({:ensure_counters_restarted, pipeline}, _, state) do
-    if Map.get(state.listeners, pipeline) do
-      Counters.start!(pipeline)
-    end
+  def handle_call(:ensure_counters_restarted, _, state) do
+    Counters.start!(state.pipeline)
 
     {:reply, :ok, state}
   end
 
   @impl true
-  def handle_call({:listen, parent, pipeline}, _, state) do
-    case Process.whereis(pipeline) do
+  def handle_call({:listen, parent}, _, state) do
+    case Process.whereis(state.pipeline) do
```

Comment:
I think you can remove this `Process.whereis` since you already check if the pipeline exists before? There is also the question of what happens if the pipeline dies... but I don't think you care, the telemetry events should still still work if they are restarted.

Reply:
For now we don't care if the pipeline dies. Maybe in the future we can monitor the pipeline and display its status. And yeah, if it restart, we are restarting the counters.